src/cascade/model/age_time_grid.py

```diff
@@ -0,0 +1,153 @@
+from datetime import timedelta
+from itertools import product
+from math import nan, inf
+
+import numpy as np
+import pandas as pd
+
+from cascade.dismod.constants import PriorKindEnum
+
+GRID_SNAP_DISTANCE = 1 / timedelta(days=365).total_seconds()
+"""Times within one second are considered equal."""
+
+
+class AgeTimeGrid:
+    """The AgeTime grid holds rows of a table at each age and time value.
+
+    At each age and time point is a dataframe consisting of the columns
+    given in the constructor. So getting an item returns a dataframe
+    with those columns. Setting a dataframe sets those columns.
+    Each AgeTimeGrid has three possible mulstds, for value, dage, dtime.
+
+    >>> atg = AgeTimeGrid(([0, 10, 20], [1990, 2000, 2010]), ["height", "weight"])
+    >>> atg[:, :] = [6.1, 195]
+    >>> atg[:, :].height = [5.9]
+    >>> atg[10, 2000] = [5.7, 180]
+    >>> atg[5:17, 1980:1990].weight = 125
+    >>> assert (atg[20, 2000].weight == 195).all()
+    >>> assert isinstance(atg[0, 1990], pd.DataFrame)
+
+    If the column has the same name as a function (mean), then access it
+    with getitem,
+
+    >>> atg[:, :]["mean"] = [5.9]
+
+    """
+    def __init__(self, age_time_grid, columns, count=1):
```

Comment:
Is there a reason for age_time_grid to be a tuple of lists rather than two arguments? It feels like that reduces clarity and makes a whole class of bugs possible that wouldn't be otherwise.

Reply:
The reason... I was thinking they are always together, but they truly aren't, and it's starting to feel stupid. Let's change it to ages and times as separate arguments.